lms/djangoapps/lti_provider/tests/test_outcomes.py

```diff
@@ -0,0 +1,399 @@
+"""
+Tests for the LTI outcome service handlers, both in outcomes.py and in tasks.py
+"""
+
+from django.test import TestCase
+from lxml import etree
+from mock import patch, MagicMock, ANY
+from opaque_keys.edx.keys import CourseKey, UsageKey
+from student.tests.factories import UserFactory
+
+from lti_provider.models import GradedAssignment, LtiConsumer, OutcomeService
+import lti_provider.outcomes as outcomes
+import lti_provider.tasks as tasks
+
+
+class StoreOutcomeParametersTest(TestCase):
+    """
+    Tests for the store_outcome_parameters method in outcomes.py
+    """
+
+    def setUp(self):
+        super(StoreOutcomeParametersTest, self).setUp()
+        course_id = 'some/course/id'
+        usage_id = 'i4x://some/course/problem/uuid'
+        self.user = UserFactory.create()
+        self.course_key = CourseKey.from_string(course_id)
+        self.usage_key = UsageKey.from_string(usage_id)
```

Comment:
I was following the examples in https://github.com/edx/edx-platform/wiki/Opaque-Keys-(Locators). What constructor would be better to call in this case?